torch/_inductor/autotuner/model.py

```diff
@@ -0,0 +1,649 @@
+                                        self.is_scalar_ln(
+                                            x[:, 60 + i * 17 + 15].long()
+                                        ),
+                                        self.is_indirect_ln(
+                                            x[:, 60 + i * 17 + 16].long()
+                                        ),
+                                    ],
+                                    dim=1,
+                                ),
+                            ],
+                            dim=1,
+                        )
+                        for i in range(15)
+                    ],
+                    dim=1,
+                ),
+                x[:, 315:],
+            ],
+            dim=1,
+        )
+        if self.use_norm:
+            for norm, layer in zip(self.norms, self.layers[:-1]):
+                x = self.activation(norm(layer(x)))
+        else:
+            for layer in self.layers[:-1]:
+                x = self.activation(layer(x))
+        x = self.layers[-1](x)
+        return x
+
+
+def get_model(model_type: ModelType):
+    if model_type == ModelType.XGB_BASELINE:
+        import xgboost
+
+        return xgboost.XGBRegressor(
+            max_depth=15,
+            learning_rate=0.2,
+            n_estimators=120,
+            tree_method="hist",
+            predictor="cpu_predictor",
+            eval_metric=["rmse", "mae"],
+        )
+    elif model_type == ModelType.NN_POINTWISE:
+        return NN(hidden_dim=[8192, 2048, 32])
+    elif model_type == ModelType.NN_PAIRWISE:
+        return NN(hidden_dim=[4096, 1024, 32])
+    elif model_type == ModelType.NN_PAIRWISE_SMALL:
+        return NN(hidden_dim=[8192, 64], use_norm=False, activation="leaky_relu")
+    else:
+        assert False, "Unknown model type"
+
+
+### feature extraction related
+
+# op_dict needs to be deterministic
+op_dict = {
+    "load": 0,
+    "to_dtype": 1,
+    "add": 2,
+    "reduction": 3,
+    "constant": 4,
+    "div": 5,
+    "store": 6,
+    "sub": 7,
+    "square": 8,
+    "rsqrt": 9,
+    "mul": 10,
+    "tanh": 11,
+    "ne": 12,
+    "where": 13,
+    "indirect_indexing": 14,
+    "log": 15,
+    "neg": 16,
+    "exp": 17,
+    "maximum": 18,
+    "minimum": 19,
+    "index_expr": 20,
+    "ge": 21,
+    "masked": 22,
+    "lt": 23,
+    "and_": 24,
+    "erf": 25,
+    "eq": 26,
+    "le": 27,
+    "gt": 28,
+    "relu": 29,
+    "sqrt": 30,
+    "logical_not": 31,
+    "load_seed": 32,
+    "rand": 33,
+    "abs": 34,
+    "reciprocal": 35,
+    "ceil": 36,
+    "sigmoid": 37,
+    "sin": 38,
+    "cos": 39,
+    "logical_and": 40,
+    "bitwise_and": 41,
+    "randn": 42,
+    "floor": 43,
+    "remainder": 44,
+    "isinf": 45,
+    "logical_or": 46,
+    "expm1": 47,
+    "libdevice_sqrt": 48,
+    "libdevice_log": 49,
+    "truediv": 50,
+    "sign": 51,
+    "randint64": 52,
+    "bitwise_or": 53,
+    "pow": 54,
+    "isnan": 55,
+}
```

Comment:
This dict is now gathered from the training data.

If we want some safety net here, likely the choice is to cover all the triton ops. But this can also shift when Triton mutates its op set.

Reply:
Yeah, I worry this will slowly become more and more out of date and nobody will ever update it.

Reply:
I don't have a good idea on this.

Currently if some op is not in the dict, it throws.